### ISSUE TITLE
basicAuth secret configuration is backwards

### ISSUE THREAD

**Issue by archSeer** [coreos/prometheus-operator#402]
The [BasicAuth section of the API docs](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#basicauth) says that [v1.SecretKeySelector scheme](https://kubernetes.io/docs/api-reference/v1.6/#secretkeyselector-v1-core) from Kubernetes is used. Based on that spec, the `name` is the name of the secret to use, and `key` is the key inside that secret. I specified it accordingly, and my metrics collection kept silently failing with no errors in any of the logs:

```
spec:
  endpoints:
  - basicAuth:
      password:
        name: prometheus-basic-auth
        key: password
      username:
        name: prometheus-basic-auth
        key: username
# ...
```

Then I came across some of the user guide examples, the `name` and `key` fields are actually reversed in prometheus-operator:

https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/basic-auth.md#basic-auth-for-targets

Swapping the key names, everything started working:

```
spec:
  endpoints:
  - basicAuth:
      password:
        key: prometheus-basic-auth
        name: password
      username:
        key: prometheus-basic-auth
        name: username
# ...
```

**Comment by brancz** [comment]
@gianrubio do you think you can have a look at this?

**Comment by gianrubio** [comment]
It's a mistake, I'll submit a PR to fix it.

**Comment by archSeer** [comment]
That will break backwards compatibility though :/

**Comment by brancz** [comment]
It's a bugfix, so it'll need to be done. The reality is that any change will always "break" someone. To upgrade properly you should use it the way it works right now and then once we release this you delete your Prometheus Operator, change the values in the TPR and deploy the new version of the Prometheus Operator. Without breaking your cluster that upgrade path will work.

**Comment by gianrubio** [comment]
@archSeer I did the PR, could you check if this fix your issue?

@brancz remember to add a good release note for this change.

**Comment by brancz** [comment]
Will do @gianrubio ! Thanks a lot! There are a couple of bug fixes pending to be released, so once that is merged we can release 0.10.1.
